packages/in_app_purchase/lib/src/in_app_purchase_connection/in_app_purchase_connection.dart

```diff
@@ -7,6 +7,118 @@ import 'dart:io';
+///
+/// on Android, all purchase information should also be verified manually, with your
+/// server if at all possible. See [`Verify a purchase`](https://developer.android.com/google/play/billing/billing_library_overview#Verify).
+class PurchaseVerificationData {
+  /// The data used for local verification.
+  ///
+  /// If the [source] is [PurchaseSource.AppStore], this data is a based64 encoded string. The structure of the payload is defined using ASN.1.
+  /// If the [source] is [PurchaseSource.GooglePlay], this data is a JSON String.
+  ///
+  /// If the platform is iOS, it is possible the data can be null or your validation of this data turns out invalid. When these happen,
+  /// Call [InAppPurchaseConnection.refreshPurchaseVerificationData] to get a new [PurchaseVerificationData] object. And then you can
+  /// validate th receipt data again using one of the methods mentioned in [`Receipt Validation`](https://developer.apple.com/library/archive/releasenotes/General/ValidateAppStoreReceipt/Introduction.html#//apple_ref/doc/uid/TP40010573-CH105-SW1).
+  final String localVerificationData;
+
+  /// The data used for server verification.
+  ///
+  /// If the platform is iOS, this data is identical to [localVerificationData].
+  final String serverVerificationData;
+
+  /// Indicates the source of the purchase.
+  final PurchaseSource source;
+
+  PurchaseVerificationData(
+      {@required this.localVerificationData,
+      @required this.serverVerificationData,
+      @required this.source});
+}
+
+/// Which platform the purchase is on.
+enum PurchaseSource { GooglePlay, AppStore }
+
+enum PurchaseStatus { pending, purchased, error }
+
+/// Error of a purchase process.
+///
+/// The error can happen during the purchase, or restoring a purchase.
+/// The error happened in restoring a purchase is not necessary the same as the error happened during the same purchase if any.
+class PurchaseError {
+  PurchaseError({@required this.source, this.code, this.message});
+
+  /// Which source is the error on.
+  final PurchaseSource source;
+
+  /// The error code.
+  final String code;
+
+  /// A map containing the detailed error message.
+  final Map<String, dynamic> message;
+}
+
+/// Represents the transaction details of a purchase.
+class PurchaseDetails {
+  /// A unique identifier of the purchase.
+  final String purchaseID;
+
+  /// The product identifier of the purchase.
+  final String productId;
+
+  /// The verification data of the purchase.
+  ///
+  /// Use this to verify the purchase.
+  final PurchaseVerificationData verificationData;
+
+  /// The timestamp of the transaction.
+  ///
+  /// Milliseconds since epoch.
+  final String transactionDate;
+
+  /// The original purchase data of this purchase.
+  ///
+  /// It is only available when this purchase is a restored purchase in iOS.
```

Comment:
Do we need to nest the originalPurchase here? I think it would be cleaner if the structure of this was the same across all our platforms.